iota/transaction/creation.py

```diff
@@ -488,3 +488,65 @@ def _create_input_transactions(self, addy):
                 # Note zero value; this is a meta transaction.
                 value=0,
             ))
+
+    def add_signature_or_message(
+            self,
+            fragments,  # type: Iterable[Fragment]
+            start_index=0  # type: Optional(int)
+    ):
+        # type: (Iterable[Fragment], int) -> None
+        """
+        Adds signature/message fragments to transactions in the bundle
+        starting at start_index.
+
+        :param fragments:
+            List of fragments to add.
+            Use [Fragment(...),Fragment(...),...] to create this argument.
+            Fragment() accepts any TryteString compatible type, or types that
+            can be converted to TryteStrings (bytearray, unicode string, etc.).
+            If the payload is less than FRAGEMENT_LENGTH, it will pad it with 9s.
+
+        :param start_index:
+            Index of transaction in bundle from where addition shoudl start.
+
+        """
+        if not isinstance(fragments, Iterable):
+            raise TypeError('fragments must be iterable')
+
+        if not all(isinstance(x, Fragment) for x in fragments):
+            raise TypeError('Iterable contains non Fragment types')
+
+        if not isinstance(start_index, int):
+            raise TypeError('start_index argument has wrong type')
+
+        length = len(fragments)
+
+        if not length:
+            raise ValueError('Empty list provided for fragments.')
+
+        if length > len(self):
+            raise ValueError('Too many fragments and too few transactions in the bundle.')
```

Comment:
❔ This is probably redundant with the `if start_index + length > len(self)` check below.

Reply:
indeed it is